### PR TITLE
Updated api-version and added DNR Changes

### PR DESCRIPTION
**Pull request by megheshpatil**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py [hunk 1]
```diff
@@ -99,13 +99,16 @@ def from_connection_string(
...
             search_id,  # type: str
+            consentdnr=False, # type: bool
             **kwargs  # type: Any
     ):
         # type: (...) -> LROPoller[None]
         """Purchases phone numbers.
 
         :param search_id: The search id.
         :type search_id: str
+        :param consent_to_not_resell_numbers/consentdnr: The consent Provided To Not Resell Phone Numbers.
```
Comment (sofiar-msft) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py:110]: nit: The consent provided to not resell phone numbers.

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py [hunk 2]
```diff
@@ -99,13 +99,16 @@ def from_connection_string(
     def begin_purchase_phone_numbers(
             self,
             search_id,  # type: str
+            consentdnr=False, # type: bool
```
Comment (sofiar-msft) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_phone_numbers_client.py:102]: I would rename it to consent_dnr

File: sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_generated/operations/_operations.py [hunk 3]
```diff
@@ -1,12 +1,12 @@
-# pylint: disable=too-many-lines
+# pylint: disable=too-many-lines,too-many-statements
```
Comment (whisper6284) [inline sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_generated/operations/_operations.py:1]: Why is this change here?

Comment (megheshpatil) [reply sdk/communication/azure-communication-phonenumbers/azure/communication/phonenumbers/_generated/operations/_operations.py]: removed the change.

File: sdk/communication/azure-communication-phonenumbers/swagger/PHONE_NUMBER_SWAGGER.md [hunk 4]
```diff
@@ -22,4 +22,18 @@ v3: true
...
+  where-model: PhoneNumberSearchResult
+  remove-property: error
+```
+
+### Removed Property errorCode from PhoneNumberSearchResult
+``` yaml
+directive:
+  where-model: PhoneNumberSearchResult
+  remove-property: errorCode
 ```
```
Comment (danielortega-msft) [inline sdk/communication/azure-communication-phonenumbers/swagger/PHONE_NUMBER_SWAGGER.md:39]: Could we add the same suppressions from the .NET SDK to prevent cherry picker generation? This is important because the generated client counts towards our coverage, but it is not used at all.

https://github.com/Azure/azure-sdk-for-net/blob/03d46a5e0a868fa42d4e25c2bd8065932a209e0e/sdk/communication/Azure.Communication.PhoneNumbers/src/autorest.md?plain=1#L44-L54
